### PR TITLE
UGENE-6686: Correct sequence "common statistics"

### PR DESCRIPTION
**Pull request by atiunov**

### CODE REVIEW

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h [hunk 1]
```diff
@@ -30,44 +30,60 @@
 
 namespace U2 {
 
+using MononucleotidesExtinctionCoefficientsMap = QVector<int>;
+using DinucleotidesExtinctionCoefficientsMap = QVector<QVector<int>>;
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h:34]: Add a space between ">"

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp [hunk 2]
```diff
@@ -31,27 +32,235 @@
...
+                        char char2 = alphabetChars[j][m];
+                        value += res[char1][char2];
+                    }
+                }
+                const int count = alphabetChars[i].length() * alphabetChars[j].length();
+                value /= count;
+                res[i][j] = value;
+            }
+        }
+    }
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp]: These lines copy-pastes the same code from the createDnaDinucleotidesExtinctionCoefficients function. I think, it's better to move it to the separate function

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp [hunk 3]
```diff
@@ -105,10 +314,36 @@ static QVector<int> createChargeMap() {
...
+    res['X'] = 0.5;
+    res['Y'] = 0.5;
+    return res;
+}
+
+const QVector<double> DNAStatisticsTask::dnaMolecularWeightMap = createDnaMolecularWeightMap();
+const QVector<double> DNAStatisticsTask::rnaMolecularWeightMap = createRnaMolecularWeightMap();
+
+const QVector<int> DNAStatisticsTask::dnaMononucleotidesExtinctionCoefficients = createDnaMononucleotidesExtinctionCoefficients();
+const QVector<QVector<int>> DNAStatisticsTask::dnaDinucleotidesExtinctionCoefficients = createDnaDinucleotidesExtinctionCoefficients();
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp]: Add a space between ">"

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp [hunk 4]
```diff
@@ -105,10 +314,36 @@ static QVector<int> createChargeMap() {
...
+const QVector<int> DNAStatisticsTask::rnaMononucleotidesExtinctionCoefficients = createRnaMononucleotidesExtinctionCoefficients();
+const QVector<QVector<int>> DNAStatisticsTask::rnaDinucleotidesExtinctionCoefficients = createRnaDinucleotidesExtinctionCoefficients();
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp]: Add a space between ">"

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h [hunk 5]
```diff
@@ -30,44 +30,60 @@
...
 
-    qint64 nA;
-    qint64 nC;
-    qint64 nG;
-    qint64 nT;
+    QVector<qint64> charactersCount;
+    QVector<qint64> rcCharactersCount;
+    QVector<QVector<qint64>> dinucleotidesCount;
+    QVector<QVector<qint64>> rcDinucleotidesCount;
+
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h:77]: Add a space between ">"

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h [hunk 6]
```diff
@@ -30,44 +30,60 @@
...
+    static const MononucleotidesExtinctionCoefficientsMap dnaMononucleotidesExtinctionCoefficients;
+    static const DinucleotidesExtinctionCoefficientsMap dnaDinucleotidesExtinctionCoefficients;
+    static const MononucleotidesExtinctionCoefficientsMap rnaMononucleotidesExtinctionCoefficients;
+    static const DinucleotidesExtinctionCoefficientsMap rnaDinucleotidesExtinctionCoefficients;
 
     static QVector<double> pMWMap; // protein molecular weight
     static QVector<double> pKaMap; // pKa values
     static QVector<int> pChargeMap; // protein charges
+    static QVector<double> gcRatioMap;  // how much contribution the character makes to the GC content
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h]: Rename static variables in a SCREAMING_SNAKE_CASE style

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h (hunk 1 above)
Comment (atiunov) [reply src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h:34]: c++11 allows two '>>'.

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp (hunk 3 above)
Comment (atiunov) [reply src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp]: c++11 allows it.

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp (hunk 4 above)
Comment (atiunov) [reply src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp]: c++11 allows it.

File: src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h (hunk 5 above)
Comment (atiunov) [reply src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.h:77]: c++11 allows it.

**Review by DmitriiSukhomlinov — APPROVED** [review]

**Pull request merged**